### PR TITLE
[exporter/elasticsearch] deprecate 'dedup' config

### PR DESCRIPTION
**Pull request by axw**
**Description:**

Deprecate `exporter/elasticsearch`'s "dedup" configuration. In a future release we will remove this configuration, and always deduplicate wherever necessary.

**Link to tracking Issue:**

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33773

**Testing:**

N/A

**Documentation:**

Updated the README.

### CODE REVIEW

File: .chloggen/elasticsearch-deprecate-dedup.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecate the "dedup" configuration.
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [33773]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (codefromthecrypt) [inline .chloggen/elasticsearch-deprecate-dedup.yaml:18]:
```suggestion
subtext: dedot has been deprecated, and will always be enabled in ECS mode in future
```

**Review by codefromthecrypt — APPROVED** [review]
thanks!

File: .chloggen/elasticsearch-deprecate-dedup.yaml (hunk 1 above)
Comment (axw) [inline .chloggen/elasticsearch-deprecate-dedup.yaml:18]:
```suggestion
subtext: dedup has been deprecated, and will always be enabled in future.
```

Comment (axw) [reply .chloggen/elasticsearch-deprecate-dedup.yaml:18]: almost, the other PR is for dedot :) adapted, thanks

**Review by codefromthecrypt — APPROVED** [review]

**Review by codefromthecrypt — APPROVED** [review]

**Review by carsonip — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**